debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/AbstractReader.java

```diff
@@ -310,7 +311,7 @@ protected void pollComplete(List<SourceRecord> batch) {
      * @throws InterruptedException if interrupted while waiting for the queue to have room for this record
      */
     protected void enqueueRecord(SourceRecord record) throws InterruptedException {
-        if (record != null) {
+        if (record != null && running.get()) {
```

Comment:
Note the reported issue still *might* happen in theory, in case a ton of records get enqueued after draining in stop and before flipping the running flag. It seems extremely unlikely, though, hence I'm prefering this over introducing yet another flag as originally suggested in #734.